load_balanced/src/main/scala/natsconnector/NatsConfig.scala

```diff
@@ -234,8 +240,50 @@ class NatsConfig(isSource: Boolean) {
       case e: NoSuchElementException =>
     }
 
+    try {
+      val param = parameters("nats.ssl.context.factory.class").toBoolean
+      this.sslContextFactoryClass = param
+    } catch {
+      case e: NoSuchElementException =>
+    }
+
+    try {
+      val param = parameters("nats.key-store.path").toBoolean
+      this.keystorePath = param
+    } catch {
+      case e: NoSuchElementException =>
+    }
+
+    try {
+      val param = parameters("nats.key-store.password").toBoolean
+      this.keystorePassword = param
+    } catch {
+      case e: NoSuchElementException =>
+    }
+
+    try {
+      val param = parameters("nats.trust-store.path").toBoolean
+      this.truststorePath = param
+    } catch {
+      case e: NoSuchElementException =>
+    }
+
+    try {
+      val param = parameters("nats.trust-store.password").toBoolean
+      this.truststorePassword = param
+    } catch {
+      case e: NoSuchElementException =>
+    }
+
+    try {
+      val param = parameters("nats.tls.algorithm").toBoolean
+      this.tlsAlgorithm = param
+    } catch {
+      case e: NoSuchElementException =>
+    }
+
     this.server = Some(s"nats://${this.host}:${this.port}")
-    this.options = Some(createConnectionOptions(this.server.get, this.allowReconnect))
+    this.options = Some(createConnectionOptions())
```

Comment:
@jnmoyne I don't understand why these values where passes as they are in the same class and you access every other param with `this.`